app/airq/models/clients.py

```diff
@@ -301,3 +305,22 @@ def maybe_notify(self) -> bool:
 
     def log_event(self, event_type: EventType, **event_data: typing.Any) -> Event:
         return Event.query.create(self.id, event_type, **event_data)
+
+    def get_last_client_event(self, exclude_type: EventType) -> Event:
+        return (
+            Event.query.filter(Event.client_id == self.id)
+            .filter(Event.type_code != exclude_type)
+            .order_by(Event.timestamp.desc())
+            .first()
+        )
+
+    def should_accept_feedback(self) -> bool:
+        last_event = self.get_last_client_event(
+            exclude_type=EventType.ALERT  # exclude alerts
+        )
+        if not last_event:
+            return False
+        return (
```

Comment:
Could compactify this to:

```python3
return (
    last_event
    and last_event.type_code == EventType.FEEDBACK_BEGIN and
    now() - last_event.timestamp < self.FEEDBACK_RESPONSE_TIME
)

Reply:
I think then it should be 

```
return bool(
    last_event
    and last_event.type_code == EventType.FEEDBACK_BEGIN and
    now() - last_event.timestamp < self.FEEDBACK_RESPONSE_TIME
)
```

to ensure it returns `False` when last_event is None

Reply:
SGTM.